plot-lib/plot/private/common/types.rkt

```diff
@@ -162,3 +162,18 @@
    [draw-pict (->* [pict (Vectorof Real)] (Anchor Real) Void)]
    [calculate-legend-rect (-> (Listof legend-entry) Rect Anchor Rect)]
    [draw-legend (-> (Listof legend-entry) Rect Void)]))
+
+(define-type Plot-Metrics<%>
+  (Class
+   [get-plot-bounds (-> (Vectorof (Vector Real Real)))]
+   [plot->dc (-> (Vectorof Real) (Vectorof Real))]))
```

Comment:
We also need a `dc->plot` method, since that would be used to convert mouse coordinates to the plot space.

Reply:
What would be the return type be for plot3D, a line?

Reply:
Or alternatively: apart from adding `dc->plot`, also provide a `plane-vector` method. For 2D this would always return #(0 0 1), for 3D this would return the unit vector `e` that is perpendicular to the screen. So for `dc->plot` 3D can just return the coordinate `c0` for the plane trough the origin. Then the user can reconstruct the right coordinate by adding these two together: `(+ c0 (* a e))`

Reply:
I did not think about the 3D case and the 2D plot area already has a `dc->plot` method.   

I am thinking about this feature in terms of "what functionality is needed to construct interactive plots?" since this was the original use case for creating #83.  I have some experience with this, from using the `plot-snip%` interface for 2D plots, and I know that there needs some way from converting from DC to plot coordinates, so be able to interact with the mouse.

For 3D plots, I have no experience with them, and not sure what is needed to achieve the final goal.  Making `dc->plot` return a line is the mathematically correct thing to do, but I am not sure if this would be sufficient and/or useful for anything practical.

Perhaps we could not do the 3D plot changes at all, and wait until someone comes with a use case that we can use to validate the design?